compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -2184,7 +2185,7 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
                     let deletion_span = deletion_span();
                     // the give lifetime originates from expanded code so we won't be able to remove it #104432
                     let lifetime_only_in_expanded_code =
-                        deletion_span.map(|sp| sp.in_derive_expansion()).unwrap_or(true);
+                        deletion_span.is_none_or(|sp| sp.in_derive_expansion());
                     if !lifetime_only_in_expanded_code {
```

Comment:
Here's another case where the condition is inverted, such that `is_some_and` might be easier to read:

```rust
                    if deletion_span.is_some_and(|sp| !sp.in_derive_expansion()) {
```